proto/ommx/v1/constraint_hint.proto

```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
+
+package ommx.v1;
+
+import "ommx/v1/one_hot.proto";
+
+// A constraint hint is an additional inforomation to be used by solver to gain performance.
+// They are derived from one-or-more constraints in the instance and typically contains information of special types of constraints (e.g. one-hot, SOS, ...).
+message ConstraintHint {
+  // The list of constraint IDs involved in the constraint.
+  repeated uint64 constraint_ids = 1;
+  oneof constraint_type {
+    // One-hot constraint: e.g. `x_1 + ... + x_n = 1` for binary variables `x_1, ..., x_n`.
+    OneHot one_hot = 2;
+  }
+}
```

Comment:
Got it. Changed in c09f14e.

Reply:
Ah, I think I fixed in a wrong way. We also should rename `ConstraintHint` to `ConstraintHints` and make `Instance` message to have one `ConstraintHints` instead of repeated - otherwise, constraints can be expressed as a nested lists.

Reply:
Fixed in [9bb8957](https://github.com/Jij-Inc/ommx/pull/180/commits/9bb895765e0325636e4b3cde421883032f8da2de).